Add UserSleepWindow to UserSettings and Allow None Type for WeatherLocation Attributes and golf_distance_unit

Jordanbarker opened the pull request:
This aims to fix issue 94: https://github.com/matin/garth/issues/94

There are two fixes:
1. Created a new dataclass `UserSleepWindow` to be created within `UserSettings`
2. Allow `None` for all `WeatherLocation` attributes and the `golf_distance_unit`.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added support for managing multiple user sleep windows in user settings.  
- **Improvements**
  - Enhanced flexibility by allowing weather location and golf distance unit settings to be left blank if desired.  
  - Expanded hydration container data to accept string values in addition to numbers.  
- **Tests**
  - Improved user settings tests and added coverage for sleep window settings.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

matin commented:
@Jordanbarker can you take a look at the failing CI?

for the tests, you'll need to record an additional VCR cassette that includes `user_sleep_windows`

Jordanbarker commented:
@matin You bet - I think It's resolved now, but I could use a closer inspection since I'm new to this testing framework. I was having difficulties getting it setup.

- Renamed `test_user_setttings` -> `test_user_settings` in a couple of places.
- Added `test_user_settings_sleep_windows.yaml` by copying `test_user_settings.yaml` and adding the key `userSleepWindows` formatted the way Garmin returns it for me. 
- Added a new pytest `test_user_settings_sleep_windows()` that is exactly the same as `test_user_settings` in test/test_users.py. Is that correct?
